truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java

```diff
@@ -1296,13 +1296,17 @@ public Object deleteAtNullOrEmpty(RubyArray array, int index) {
         private final BranchProfile nextProfile = BranchProfile.create();
         private final BranchProfile redoProfile = BranchProfile.create();
 
+        private final RubySymbol eachSymbol;
+
         public EachNode(RubyContext context, SourceSection sourceSection) {
             super(context, sourceSection);
+            eachSymbol = getContext().getSymbolTable().getSymbol("each");
```

Comment:
Didn't know about that. I would expect it to be `getContext().getSymbol()` since this will likely not create a new Symbol everytime.